subgraphs/notional-finance/src/setters/financialMetrics.ts

```diff
@@ -26,55 +25,64 @@ import {
 } from "../getters/market";
 import { getOrCreateLendingProtocol } from "../getters/protocol";
 import { getOrCreateToken } from "../getters/token";
+import { getOrCreateInterestRate } from "../getters/interestRate";
+import { MarketDailySnapshot } from "../../generated/schema";
 
 export function updateFinancials(
   event: ethereum.Event,
-  amountUSD: BigDecimal,
-  marketId: string
+  marketId: string,
+  prevMarketDailySnapshot: MarketDailySnapshot
 ): void {
-  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
-  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, marketId);
-  const marketDailySnapshot = getOrCreateMarketDailySnapshot(event, marketId);
-  const protocol = getOrCreateLendingProtocol();
   const market = getOrCreateMarket(event, marketId);
+  const protocol = getOrCreateLendingProtocol();
+  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, marketId);
+  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
 
-  // fees and revenue amounts
-  const feesUSD = amountUSD.times(NOTIONAL_TRADE_FEES);
-  const totalRevenueUSD = feesUSD;
-  const supplySideRevenueUSD = feesUSD.times(
-    NOTIONAL_SUPPLY_SIDE_REVENUE_SHARE
+  const dailyId = event.block.timestamp.toI64() / SECONDS_PER_DAY;
+  const marketDailySnapshot = getOrCreateMarketDailySnapshot(
```

Comment:
Kind of a strange loop where you call `updateFinancials()` from `getOrCreateMarketDailySnapshot()` then you call `getOrCreateMarketDailySnapshot()` in `updateFinancials()`... Is this intentional?

Reply:
Yes, this is intentional.

There is no direct circular dependency between `updateRevenues` (previously updateFinancials) and `getOrCreateMarketDailySnapshot`.
* The `updateRevenues` calls getOrCreateMarketDailySnapshot internally to obtain the `marketDailySnapshot` entity for updating its revenue-related fields. This indirect dependency helps in calculating the daily and cumulative revenues for the market.
* Whereas the `getOrCreateMarketDailySnapshot` calls the updateRevenues function when creating a `marketDailySnapshot` to calculate and update the revenues based on the previous metrics if a previous MarketDailySnapshot exists.

I also don't like this approach but was required to update metrics of previous daily market snapshot when new daily market snapshot was being created. Exploring an alternative approach, will push soon.

Reply:
Yes, this is intentional.

There is no direct circular dependency between `updateRevenues` (previously updateFinancials) and `getOrCreateMarketDailySnapshot`.
* The `updateRevenues` calls getOrCreateMarketDailySnapshot internally to obtain the `marketDailySnapshot` entity for updating its revenue-related fields. This indirect dependency helps in calculating the daily and cumulative revenues for the market.
* Whereas the `getOrCreateMarketDailySnapshot` calls the updateRevenues function when creating a `marketDailySnapshot` to calculate and update the revenues based on the previous metrics if a previous MarketDailySnapshot exists.

I also don't like this approach but was required to update metrics of previous daily market snapshot when new daily market snapshot was being created. Exploring an alternative approach, will push soon.